src/equations/compressible_euler_1d.jl

```diff
@@ -108,8 +111,8 @@ Source terms used for convergence tests in combination with
     # Note that d/dt rho = -d/dx rho.
     # This yields du2 = du3 = d/dx p (derivative of pressure).
     # Other terms vanish because of v = 1.
-    du1 = zero(eltype(u))
-    du2 = rho_x * (2 * rho - 0.5) * (γ - 1)
+    du1 = 0
```

Comment:
The fact that `zero(eltype(u))` or `one(eltype(u))` is (sometimes) not necessary in these cases is also documentation-worthy imho

Reply:
This is something we can discuss. Using `zero(eltype(u))` or `one(eltype(u))` in cases like this will make the code longer. It's not necessary here since the `SVector` constructor takes care of the type promotion later. However, it may not be clear which choice is easier to read/remember in the end - "just use `zero`/`one` everywhere" or "check whether it will be promoted later and use `0`/`1` if so" ?

Reply:
It would be ok for me to leave it up to the implementer to decide how to proceed, but this should also be put in the docs. That is, one can decide to take the safe-but-more-verbose route, or the concise-but-more-advanced solution.